storage/pebble/registers_cache.go

```diff
@@ -0,0 +1,83 @@
+package pebble
+
+import (
+	"fmt"
+
+	"github.com/pkg/errors"
+
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/module"
+	"github.com/onflow/flow-go/storage"
+)
+
+const (
+	registerResourceName = "registers"
+)
+
+type RegistersCache struct {
+	registers *Registers
+	cache     *ReadCache
+}
+
+var _ storage.RegisterIndex = (*RegistersCache)(nil)
+
+// NewRegistersCache wraps a read cache around Get requests to a underlying Registers object.
+func NewRegistersCache(registers *Registers, cacheType CacheType, size uint, metrics module.CacheMetrics) (*RegistersCache, error) {
+	if size == 0 {
+		return nil, errors.New("cache size cannot be 0")
+	}
+
+	cache, err := newReadCache(
```

Comment:
The responsibility of creating a cache doesnt need to be in here... 
The new function could just be: `NewRegistersCache(registers *Registers, cache     *ReadCache)`

Reply:
I don't fell strongly about this.

Reply:
ok I'll leave it for now. I think we can evolve this some more to make the API more general purpose as we add more types to pebble.